utils/BuildPackages.ps1

```diff
@@ -0,0 +1,171 @@
+    [string]$storageAccount,
+
+    # The Azure account key to use.
+    [string]$storageKey,
+
+    # The Azure container to use.
+    [string]$storageContainer,
+
+    # The path to NuGet. Defaults to "nuget.exe".
+    [string]$nugetPath = "nuget.exe",
+
+    # The output directory for the cross-platform binaries.
+    [string]$binariesDir,
+
+    # The package output directory.
+    [string]$packageOutputDir,
+
+    # The directory that contains the .nuspec files that will be used to create the
+    # cross-platform and combined packages.
+    [string]$nuspecDir
+)
+
+function Get-Latest-Blob-Name
+{
+    Param([array]$blobs, [string]$expectedSuffix)
+
+    $chosenBlob = $null
+    $chosenDate = $null
+    foreach ($blob in $blobs)
+    {
+        if ($blob.name -notlike "*$expectedSuffix")
+        {
+            continue
+        }
+
+        $date = [datetime]($blob.properties."last-modified")
+        if (!$chosenBlob -or ($chosenDate -and $date -ge $chosenDate))
+        {
+            $chosenBlob = $blob.name
+            $chosenDate = $date
+        }
+    }
+
+    return $chosenBlob
+}
+
+# Get the list of blobs in storage
+$json = (azure storage blob list -a $storageAccount -k $storageKey $storageContainer --json) -join ""
+$blobs = ConvertFrom-Json $json
+
+# Find, fetch, and extract the latest Ubuntu and OSX blobs
+[System.Reflection.Assembly]::LoadWithPartialName("System.IO.Compression.FileSystem") | Out-Null
+
+$ubuntuBlob = Get-Latest-Blob-Name $blobs "Ubuntu.14.04_LLILC_x64_Release_Enu.zip"
+$osxBlob = Get-Latest-Blob-Name $blobs "OSX_LLILC_x64_Release_Enu.zip"
+
+if (!$ubuntuBlob)
+{
+    Write-Error "Could not locate an Ubuntu drop in Azure."
+    exit 1
+}
+
+if (!$osxBlob)
+{
+    Write-Error "Could not locate an OS X drop in Azure."
+    exit 1
+}
+
+azure storage blob download -m -q -a $storageAccount -k $storageKey $storageContainer $ubuntuBlob
+if ($LastExitCode -ne 0)
+{
+    Write-Error "Failed to fetch Ubuntu drop $ubuntuDrop from Azure."
+    exit 1
+}
+
+azure storage blob download -m -q -a $storageAccount -k $storageKey $storageContainer $osxBlob
+if ($LastExitCode -ne 0)
+{
+    Write-Error "Failed to fetch OS X drop $osxBlob from Azure."
+    exit 1
+}
+
+$ubuntuDirectory = [System.IO.Path]::GetFileNameWithoutExtension($ubuntuBlob)
+try
+{
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($ubuntuBlob, $ubuntuDirectory)
+}
+catch
+{
+    Write-Error "Failed to extract Ubuntu drop to $ubuntuDirectory`: $_.Exception.Message"
+    exit 1
+}
+
+$osxDirectory = [System.IO.Path]::GetFileNameWithoutExtension($osxBlob)
+try
+{
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($osxBlob, $osxDirectory)
+}
+catch
+{
+    Write-Error "Failed to extract OS X drop to $osxDirectory`: $_.Exception.Message"
+    exit 1
+}
+
+# Gather the bits from the Ubuntu and OSX blobs into the bin directory
+$items = @(
+    "$ubuntuDirectory\libobjectwriter.so",
+    "$osxDirectory\libobjectwriter.dylib",
```

Comment:
In some reason, we named 'objwriter' instead of "objectwriter" for the file name. Either we should fix this or update the object file name itself in our build.